src/OneExplorer/OneExplorer.ts

```diff
@@ -229,6 +229,37 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
         });
   }
 
+  delete(oneNode: OneNode): void {
+    if (oneNode.node.type === NodeType.directory) {
+      vscode.window
+          .showInformationMessage(
+              `Are you sure you want to delete '${
+                  path.parse(oneNode.node.path).base}' and its contents?`,
+              {detail: `You can restore this file from the Trash.`, modal: true}, 'Move to Trash')
+          .then(ans => {
+            if (ans === 'Move to Trash') {
+              Logger.debug('OneExplorer', `Delete '${path.parse(oneNode.node.path).base}'.`);
+              return vscode.workspace.fs.delete(
+                  oneNode.node.uri, {recursive: true, useTrash: true});
+            }
+          })
+          .then(() => this.refresh());
+    } else {
+      vscode.window
+          .showInformationMessage(
+              `Are you sure you want to delete '${path.parse(oneNode.node.path).base}'?`,
+              {detail: `You can restore this file from the Trash.`, modal: true}, 'Move to Trash')
+          .then(ans => {
+            if (ans === 'Move to Trash') {
+              Logger.debug('OneExplorer', `Delete '${path.parse(oneNode.node.path).base}'.`);
+              return vscode.workspace.fs.delete(
+                  oneNode.node.uri, {recursive: false, useTrash: true});
+            }
+          })
```

Comment:
can we define a function to reduce code duplication?
for example,
```
function confirmDeletion(uri: URI, isDirectory) {
  let title = isDirectory? " ...  and its contents?" : " ..."';
  let recursive = isDirectory? true : false;
  // showinformation code below
  ...
}
```

Reply:
Actually, I tried it but eventually the current one looked better... but well, I will try once again.

Reply:
@hyunsik-yoon PTAL,